packages/host/app/components/matrix/register-user.gts

```diff
@@ -295,62 +369,56 @@ export default class RegisterUser extends Component<Signature> {
         type: 'login';
         username: string;
         password: string;
+        email: string;
+        name: string;
         session: string;
       } = { type: 'initial' };
 
   @service private declare matrixService: MatrixService;
 
-  constructor(owner: Owner, args: Signature['Args']) {
-    super(owner, args);
-    let validationStr = localStorage.getItem('email-validation');
-    if (validationStr) {
-      let { email, sid, clientSecret, sendAttempt, session } = JSON.parse(
-        validationStr,
-      ) as Validation;
-
-      this.state = {
-        type: 'askForUserCreds',
-        email,
-        sid,
-        clientSecret,
-        session,
-        sendAttempt,
-      };
-
-      this.checkEmailValidation.perform();
-    }
-  }
-
-  private get showEmailValidationStatus() {
-    return [
-      'askForUserCreds',
-      'register',
-      'askForToken',
-      'sendToken',
-      'waitForEmailValidation',
-    ].includes(this.state.type);
-  }
-
   private get isRegisterButtonDisabled() {
-    return !this.username || !this.password || !this.confirmPassword;
-  }
-
-  private get isValidateButtonDisabled() {
-    return !this.email;
+    return (
+      !this.email ||
+      !this.name ||
+      !this.username ||
+      !this.password ||
+      !this.confirmPassword ||
+      this.emailError ||
+      this.nameError ||
+      this.usernameError ||
+      this.passwordError ||
+      this.confirmPasswordError
```

Comment:
this is hard to reason about. i think what you mean to say is that if there is a missing field or there is an error then the register button is disabled. can we break this down into a getter `get hasMissingField()` and `get hasError()` so that it is more clear what the intent is here?